src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java

```diff
@@ -0,0 +1,182 @@
+package jenkins.plugins.logstash;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.annotation.CheckForNull;
+
+import org.kohsuke.stapler.StaplerRequest;
+
+import com.cloudbees.syslog.MessageFormat;
+
+import org.apache.http.client.utils.URIBuilder;
+import hudson.Extension;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import jenkins.model.GlobalConfiguration;
+import jenkins.plugins.logstash.LogstashInstallation.Descriptor;
+import jenkins.plugins.logstash.configuration.ElasticSearch;
+import jenkins.plugins.logstash.configuration.LogstashIndexer;
+import jenkins.plugins.logstash.configuration.RabbitMq;
+import jenkins.plugins.logstash.configuration.Redis;
+import jenkins.plugins.logstash.configuration.Syslog;
+import jenkins.plugins.logstash.persistence.LogstashIndexerDao;
+import jenkins.plugins.logstash.persistence.LogstashIndexerDao.IndexerType;
+import net.sf.json.JSONObject;
+
+@Extension
+public class LogstashConfiguration extends GlobalConfiguration
+{
+  private static final Logger LOGGER = Logger.getLogger(LogstashConfiguration.class.getName());
+  private LogstashIndexer<?> logstashIndexer;
+  private boolean dataMigrated = false;
+  private transient LogstashIndexer<?> activeIndexer;
+
+  public LogstashConfiguration()
+  {
+    load();
+    if (logstashIndexer != null)
```

Comment:
Why not always assign? I don't see how `activeIndexer` could be non-null here

Reply:
It can be null when you install the plugin. But the activeIndexer being null is not a problem